### PR TITLE
Improve nightly tests

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Improve nightly tests by:

* Following a similar strategy as the push tests by running pipeline tests only on a selected set of pipelines (based on usage).
* Parallelize where applicable. 
* General improvements.

### CODE REVIEW

File: .github/workflows/nightly_tests.yml [hunk 1]
```diff
@@ -12,110 +12,96 @@ env:
...
       - name: NVIDIA-SMI
-        if: ${{ matrix.config.runner == 'docker-gpu' }}
-        run: |
-          nvidia-smi
-
+        run: nvidia-smi
+      
       - name: Install dependencies
         run: |
+          apt-get update && apt-get install libsndfile1-dev libgl1 -y
```
Comment (DN6) [inline .github/workflows/nightly_tests.yml:72]: Actually, I think we should move this step into the container building workflow.

File: .github/workflows/nightly_tests.yml [hunk 2]
```diff
@@ -124,6 +110,206 @@ jobs:
...
+    runs-on: docker-gpu
+    container:
+      image: diffusers/diffusers-pytorch-cuda
+      options: --shm-size "16gb" --ipc host -v /mnt/hf_cache:/mnt/cache/ --gpus 0
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        module: [models, schedulers, others, lora, examples]
```
Comment (DN6) [inline .github/workflows/nightly_tests.yml:124]: Can remove `lora` from this

File: .github/workflows/nightly_tests.yml [hunk 3]
```diff
@@ -124,6 +110,206 @@ jobs:
           pip install slack_sdk tabulate
           python scripts/log_reports.py >> $GITHUB_STEP_SUMMARY
 
+  run_nightly_tests_for_other_torch_modules:
+    name: Torch Non-Pipelines CUDA Nightly Tests
```
Comment (DN6) [inline .github/workflows/nightly_tests.yml:114]: I think we can just call it "Torch CUDA Nightly Tests"

**Review by DN6 — COMMENTED** [review]
Looks nice! Just a few small things to update IMO

File: .github/workflows/nightly_tests.yml (hunk 3 above)
Comment (sayakpaul) [reply .github/workflows/nightly_tests.yml:114]: I wanted to specifically distinguish between pipeline and non-pipeline tests for convenience.

File: .github/workflows/nightly_tests.yml (hunk 1 above)
Comment (sayakpaul) [reply .github/workflows/nightly_tests.yml:72]: In an immediate follow-up PR?

**Comment by sayakpaul** [comment]
@DN6 addressed your feedback. Rigorous review is appreciated.

**Comment by sayakpaul** [comment]
@DN6 a gentle ping here.

**Review by DN6 — APPROVED** [review]
Nicely done!

**Pull request merged**